### ISSUE TITLE
Persistent Storage uses caches/document directory

### ISSUE THREAD

**Issue by tcldr** [amplitude/Amplitude-Swift#104]
## Expected Behavior
Application data should be stored in the application support directory reachable via `FileManager.SearchPathDirectory.applicationSupportDirectory`.

## Current Behavior
Application data is being saved to either the user's document directory or the caches directory. The document's directory is less than ideal as it clutters up user space. The caches directory is also problematic as it can be purged by the system.

## Possible Solution
Use the application support directory (`FileManager.SearchPathDirectory.applicationSupportDirectory`). On macOS, it's the `~/Library/Application Support` directory. On iOS and other platforms it's app specific, but serves the same purpose.

Furthermore, you can mark a directory/file to be excluded from iCloud backup by setting attributes at the point of creation. More info here: [Mark Nonpurgeable Data as Excludable](https://developer.apple.com/documentation/foundation/optimizing_your_app_s_data_for_icloud_backup/#3928527)

I've created a PR you can reference

**Comment by tcldr** [comment]
Hi guys, any update on this? @justin-fiedler

**Comment by rakefetWorkiz** [comment]
it's been almost a year since this ticket was opened
any update?

**Comment by z3DD3r** [comment]
Looks like this issue was fixed with https://github.com/amplitude/Amplitude-Swift/pull/400

**Comment by crleona** [comment]
Yes, this should be resolved with #400 - Thank you for updating these issues @z3DD3r .
